web/models/setting/settings.ex

```diff
@@ -0,0 +1,79 @@
+defmodule CoursePlanner.Settings do
+  @moduledoc """
+  This module provides custom functionality for controller over the model
+  """
+  use CoursePlanner.Web, :model
+
+  alias CoursePlanner.{Repo, SystemVariable}
+  alias Ecto.Multi
+
+  @all_query               from SystemVariable, order_by: :key
+  @visible_settings_query  from sv in SystemVariable, where: sv.visible  == true, order_by: :key
+  @editable_settings_query from sv in SystemVariable, where: sv.editable == true, order_by: :key
+
+  def all do
+    Repo.all(@all_query)
+  end
+
+  def get_visible_systemvariables do
+    Repo.all(@visible_settings_query)
+  end
+
+  def get_editable_systemvariables do
+    Repo.all(@editable_settings_query)
+  end
+
+  def get_changesets_for_update(setting_params) do
+    system_variables = all()
+    setting_params = Enum.sort_by(setting_params, &(elem(&1, 1)["key"]))
+
+    Enum.map(setting_params, fn(setting_param) ->
+        {param_id, %{"key" => _param_key, "value" => param_value}} = setting_param
+
+      found_system_variable =
+        Enum.find(system_variables, fn(system_variable) ->
```

Comment:
I am not sure if I get your point correctly. But in case you are talking about "setting_params" then they are coming from the form or if it is about "system_variables" then it's directly from database